docs/serviceprincipal.md

```diff
@@ -25,12 +25,32 @@ There are several ways to create a Service Principal in Azure Active Directory:
    az ad sp create-for-rbac --role="Contributor" --scopes="/subscriptions/${SUBSCRIPTION_ID}"
    ```
 
-This will output your `client_id`, `client_secret` (password), `sp_name`, and `tenant`.  The `sp_name` or `client_id` may be used for the `servicePrincipalProfile.servicePrincipalClientId` and the `client_secret` is used for `servicePrincipalProfile.servicePrincipalClientSecret`.
+You can set `SUBSCRIPTION_ID` like this:
+
+```
+SUBSCRIPTION_ID=$(az account list --query "[?name == '<i>subscription name</i>'].[id]" --out tsv)
+```
+
+With CLI output format set to JSON, the output will look something like this:
+
+```
+{
+  "appId": "<i>some GUID</i>",
+  "name": "http://azure-cli-2016-<i>more characters</i>",
+  "password": "<i>password GUID</i>",
+  "tenant": "<i>tenant GUID</i>"
+}
+
+```
+
+The `appId` is the Azure AD `client_id`, the `password' is the `client_secret`, `the `name` is the `service principal name`, and `tenant` is the GUID identifiying the tenant in which the service principal was created.  The `service principal name` or `client_id` may be used for the `servicePrincipalProfile.servicePrincipalClientId` and the `password` or `client_secret` is used for `servicePrincipalProfile.servicePrincipalClientSecret`.

```

Comment:
I like what you're doing here but can we change it to be slightly less confusing. I think there's value in showing what `appId` and `password` map to in normal OAuth terms, but let's stick with AAD nomenclature otherwise:

---

The `appId` output maps to the `client_id`, and `password` maps to `client_secret` in normal OAuth terms. The `name` is simply a more human readable identifier for the Active Directory Application created, and the `tenant` field contains the GUID for the tenant that the new Service Principal belongs to.

The `appId` or `name` values may be used for the `servicePrincipalProfile.servicePrincipalClientId`. Similarly, the `password` value should be used for `servicePrincipalProfile.servicePrincipalClientSecret`.

---

Alternatively, I wish we would have just had this output `client_id`, `client_secret`, `identifier_uri` and `tenantId` so that these were more self-describing.